azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -445,18 +455,26 @@ private Session createSession(final HttpServletRequest req)
     final String password = getParam(req, "password");
     final String ip = WebUtils.getRealClientIpAddr(req);
 
-    return createSession(username, password, ip);
+    return createSession(username, password, ip, true);
   }
 
-  private Session createSession(final String username, final String password, final String ip)
+  private Session createSession(final String username, final String password, final String ip,
+      final boolean isSuccessFinal)
       throws UserManagerException {
-    final UserManager manager = getApplication().getUserManager();
-    final User user = manager.getUser(username, password);
+    try {
+      final UserManager manager = getApplication().getUserManager();
+      final User user = manager.getUser(username, password);
 
-    final String randomUID = UUID.randomUUID().toString();
-    final Session session = new Session(randomUID, user, ip);
+      if (isSuccessFinal) {
+        WebUtils.reportLoginEvent(EventType.USER_LOGIN, username, ip);
+      }
 
-    return session;
+      final String randomUID = UUID.randomUUID().toString();
+      return new Session(randomUID, user, ip);
+    } catch (Exception e) {
```

Comment:
I would agree with you if the exception was actually handled here -- then yes, being explicit about which exception types are handled is important. 
But in this case I explicitly want to catch **everything**, including any unchecked exceptions, because (1) we must report a failure on *any* error, and (2) we re-throw that exception (as opposed to handling and swallowing it) anyway.
Does it make sense?